cli/pkg/k8s/agent.go

```diff
@@ -15,31 +15,29 @@ const (
 	// VersionLabel is the label key for the agent's version
 	VersionLabel = "app.kubernetes.io/version"
 
-	agentSecret = "buoyant-cloud-id"
+	agentMetadataMap     = "agent-metadata"
+	orgCredentialsSecret = "buoyant-cloud-org-credentials"
 )
 
 // Agent represents the Buoyant Cloud agent. Any of these fields may not be
 // present, depending on which resources are already on the cluster.
 type Agent struct {
 	Name    string
+	Id      string
 	Version string
-	URL     string
 }
```

Comment:
this feels similar to `AgentIdentifier`, but i suppose `AgentIdentifier` is intended for input/query, and `Agent` is returned by bcloud.

Reply:
The difference is that `AgentIdentifier` is always a oneOf, while with `Agent` we can have both the fields set.  I think we can still reuse it but do not see much of a point.